### PR TITLE
Migration

### PR DESCRIPTION
**Pull request by schuemie**
This implements migration to the new results data model.

Currently requiring SqlRender 1.16.0, which hasn't been released yet, for the support of `ALTER TABLE ... ADD ...`

The final (migrated) database schema will have four fields (`outcome_rate`_ `adjusted_rate`, `stable`, and `p`) in the `time_trend` table that are not in the CSV files produced by `exportToCsv()` nor in the CSV file describing the results schema. These are deprecated, and are only there so we don't throw away information.

### CODE REVIEW

**Review by azimov — CHANGES_REQUESTED** [review]
The only really important change is the version number in the UpdateVersion.sql - which is very optional and is just something added to CohortDiagnostics.

File: inst/sql/UpdateVersionNumber.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{DEFAULT @package_version = package_version}
+{DEFAULT @version_number = '3.2.3'}
```
[the remark this answers is not in the document]
Comment (schuemie) [reply inst/sql/UpdateVersionNumber.sql:2]: But this is just the default value for SqlRender. The preferred way would be to set the `@version_number` parameter when calling `SqlRender::render()` (which I was assuming is what happened). I'm not sure why you would modify the SQL file itself. That is exactly what SqlRender is for, right?

I added default values to SqlRender just for debugging, not to actually use them in production.

File: inst/sql/CreateResultsDataModel.sql [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+{DEFAULT @sccs_likelihood_profile = sccs_likelihood_profile}
+{DEFAULT @sccs_time_trend = sccs_time_trend}
+{DEFAULT @sccs_time_to_event = sccs_time_to_event}
+{DEFAULT @sccs_age_spanning = sccs_age_spanning}
+{DEFAULT @sccs_calendar_time_spanning = sccs_calendar_time_spanning}
+{DEFAULT @sccs_event_dep_observation = sccs_event_dep_observation}
+{DEFAULT @sccs_diagnostics_summary = sccs_diagnostics_summary}
+  
+CREATE TABLE @database_schema.@table_prefix@sccs_analysis (
+  	 analysis_id INT NOT NULL,
```
[the remark this answers is not in the document]
Comment (schuemie) [reply inst/sql/CreateResultsDataModel.sql:22]: Yes, this SQL is completely generated by `ResultModelManager::generateSqlSchema` using the original CSV file here: https://github.com/OHDSI/SelfControlledCaseSeries/blob/v4.2.0/inst/csv/resultsDataModelSpecification.csv . Let me try to regenerate and see if I can find our what is happening.

[the remark this answers is not in the document]
Comment (schuemie) [reply inst/sql/CreateResultsDataModel.sql:22]: Found the problem: I'm using 'Yes' and 'No', but `generateSqlSchema()` is looking for 'yes' and 'no' (notice the case). May want to make that case insensitive.

**Review by azimov — APPROVED** [review]

**Pull request merged**